components/consents/api/src/main/java/org/phenotips/consents/internal/PhenoTipsPatientConsentManager.java

```diff
@@ -163,16 +187,49 @@ private Consent fromXWikiConsentConfiguration(BaseObject xwikiConsent, XWikiDocu
             boolean required = intToBool(xwikiConsent.getIntValue("required"));
             boolean affectsFields = intToBool(xwikiConsent.getIntValue("affectsFields"));
             List<String> formFields = null;
+            List<String> dataFields = null;
             if (affectsFields) {
-                formFields = xwikiConsent.getListValue("fields");
+                dataFields = new LinkedList<>();
+                formFields = xwikiConsent.getListValue(FIELDS);
+                fetchConsentFields(formFields, dataFields, xwikiConsent);
             }
-            return new DefaultConsent(id, label, description, required, formFields);
+            return new DefaultConsent(id, label, description, required, dataFields, formFields);
         } catch (Exception ex) {
             this.logger.error("A patient consent is improperly configured: {}", ex.getMessage());
         }
         return null;
     }
 
+    private void fetchConsentFields(List<String> formFields, List<String> dataFields, BaseObject xwikiConsent)
```

Comment:
I don't think you're using xwikiConsent anywhere in the method.

Reply:
indeed